scripts/ecnf/import_utils.py

```diff
@@ -0,0 +1,293 @@
+        allisogdegs = data1['isogeny_degrees']
+        ncurves = len(allisogdegs)
+        for n in range(ncurves):
+            isogdegs = allisogdegs[n+1]
+            label = class_label+str(n+1)
+            data[label] = {'isogeny_degrees': isogdegs,
+                           'class_size': ncurves,
+                           'class_deg': maxdeg}
+            if n==0:
+                print("adding isogmat = {} to {}".format(isogmat,label))
+                data[label]['isogeny_matrix'] = isogmat
+
+    return data
+
+
+def galrep(line):
+    r""" Parses one line from a galrep file.  Returns the label and a
+    dict containing two fields: 'non-surjective_primes', a list of
+    primes p for which the Galois representation modulo p is not
+    surjective (cut off at p=37 for CM curves for which this would
+    otherwise contain all primes), 'galois_images', a list of strings
+    encoding the image when not surjective, following Sutherland's
+    coding scheme for subgroups of GL(2,p).  Note that these codes
+    start with a 1 or 2 digit prime followed a letter in
+    ['B','C','N','S'].
+
+    Input line fields (4+); the first is a standard label of the form
+    field-conductor-an where 'a' is the isogeny class (one or more
+    letters), 'n' is the number ofe the curve in the class (from 1)
+    and any remaining ones are galrep codes.
+
+    label codes
+
+    Sample input line (field='2.0.3.1', conductor='10000.0.100', class='a', number=1)
+
+    2.0.3.1-10000.0.100-a1 2B 3B[2]
+
+    """
+    data = split(line)
+    label = data[0] # single string
+    image_codes = data[1:]
+    pr = [ int(s[:2]) if s[1].isdigit() else int(s[:1]) for s in image_codes]
+    return label, {
+        'non-surjective_primes': pr,
+        'galois_images': image_codes,
+    }
+
+
+
+
+def readgalreps(base_path, filename):
+    h = open(os.path.join(base_path, filename))
+    print("opened {}".format(os.path.join(base_path, filename)))
+    dat = {}
+    for L in h.readlines():
+        lab, dat1 = galrep(L)
+        dat[lab] = dat1
+    return dat
+
+def make_curves_line(ec):
+    r""" for ec a curve object from the database, create a line of text to
+    match the corresponding raw input line from a curves file.
+
+    Output line fields (13):
+
+    field_label conductor_label iso_label number conductor_ideal conductor_norm a1 a2 a3 a4 a6 cm base_change
+
+    Sample output line:
+
+    2.0.4.1 65.18.1 a 1 [65,65,-4*w-7] 65 1,1 1,1 0,1 -1,1 -1,0 0 0
+    """
+    cond_lab = ec['conductor_label']
+    output_fields = [ec['field_label'],
+                     cond_lab,
+                     ec['iso_label'],
+                     str(ec['number']),
+                     ec['conductor_ideal'],
+                     str(ec['conductor_norm'])
+                     ] + [",".join(t) for t in ec['ainvs']
+                          ] + [str(ec['cm']), '?' if ec['base_change']=='?' else str(int(len(ec['base_change']) > 0))]
```

Comment:
hi @JohnCremona 
The `make_curves_line` is the only function I modified. I changed it so that it prints a '?' if `ec['base_change']` is set to '?'.